components/datetime/src/time_zone.rs

```diff
@@ -251,59 +387,246 @@ impl TimeZoneFormat {
+        self.format_units
+            .push(TimeZoneFormatUnit::SpecificNonLocationLong(
+                SpecificNonLocationLongFormat {},
+            ));
+        Ok(self)
+    }
+
+    /// Load specific non location short format for timezone. For example, PDT.
+    pub fn load_specific_non_location_short<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+    where
+        ZP: ResourceProvider<provider::time_zones::MetaZoneSpecificNamesShortV1Marker> + ?Sized,
+    {
+        if self.data_payloads.mz_specific_short == None {
+            load_resource(
+                &self.locale,
+                &mut self.data_payloads.mz_specific_short,
+                zone_provider,
+            )?;
+        }
+        self.format_units
+            .push(TimeZoneFormatUnit::SpecificNonLocationShort(
+                SpecificNonLocationShortFormat {},
+            ));
+        Ok(self)
+    }
+
+    /// Load generic location format for timezone. For example, Los Angeles Time.
+    pub fn load_generic_location_format<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+    where
+        ZP: ResourceProvider<provider::time_zones::ExemplarCitiesV1Marker> + ?Sized,
+    {
+        if self.data_payloads.exemplar_cities == None {
+            load_resource(
+                &self.locale,
+                &mut self.data_payloads.exemplar_cities,
+                zone_provider,
+            )?;
+        }
+        self.format_units.push(TimeZoneFormatUnit::GenericLocation(
+            GenericLocationFormat {},
+        ));
+        Ok(self)
+    }
+
+    /// Load exemplar city format for timezone. For example, Los Angeles.
+    fn load_exemplar_city_format<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+    where
+        ZP: ResourceProvider<provider::time_zones::ExemplarCitiesV1Marker> + ?Sized,
+    {
+        if self.data_payloads.exemplar_cities == None {
+            load_resource(
+                &self.locale,
+                &mut self.data_payloads.exemplar_cities,
+                zone_provider,
+            )?;
+        }
+        self.format_units
+            .push(TimeZoneFormatUnit::ExemplarCity(ExemplarCityFormat {}));
+        Ok(self)
+    }
+
+    /// Load localized GMT format for timezone. For example, GMT-07:00.
+    pub fn load_localized_gmt_format(
+        &mut self,
+        is_fallback: bool,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError> {
+        if is_fallback {
+            self.fallback_unit = Some(TimeZoneFormatUnit::LocalizedGmt(LocalizedGmtFormat {}));
+        } else {
+            self.format_units
+                .push(TimeZoneFormatUnit::LocalizedGmt(LocalizedGmtFormat {}));
+        }
+        Ok(self)
+    }
+
+    /// Load Iso8601 format for timezone. For example, -07:00.
+    pub fn load_iso_8601_format(
```

Comment:
This seems fine to me. 

This function doesn't literally "load" any data, but it does insert this formatting option into the vec, or sets it as the fallback type. 

I'm fine with this function being called "load" just to be congruent with the other functions that do the same thing. 

We could change the terminology of all of them to be something like "add" instead of "load".

Reply:
the method is responsible to handle both fallback and non-fallback cases for `iso_8601_format`, which is to insert the format unit in the right vec.

Reply:
My question was, if `fallback_unit` handles this format, then why should it ever need to be added to the chain? It's always the final fallback unit.

Reply:
I updated the code and removed the fallback related logic in this method.